starter_kits/Ruby/hlt/map.rb

```diff
@@ -0,0 +1,138 @@
+require 'direction'
+require 'position'
+require 'cell'
+
+
+# The game map.
+# Can be indexed by a position, or by a contained entity.
+# Coordinates start at 0. Coordinates are normalized for you.
+class Map
+
+  def initialize(cells, width, height)
+    @width = width
+    @height = height
+    @cells = cells
+  end
+
+  # Getter for position object or entity objects within the game map
+  # :param location: the position or entity to access in this map
+  # :return: the contents housing that cell or entity
+  def [](location)
+    if location.is_a? Position
+      location = normalize(location)
+      return @cells[location.y][location.x]
+    elsif location.is_a? Entity
+      return @cells[location.position.y][location.position.x]
+    end
+    return nil
+  end
+
+  # Compute the Manhattan distance between two locations.
+  #   Accounts for wrap-around.
+  # :param source: The source from where to calculate
+  # :param target: The target to where calculate
+  # :return: The distance between these items
+  def calculate_distance(source, target)
+    source = normalize(source)
+    target = normalize(target)
+    resulting_position = (source - target).abs
+    return min(resulting_position.x, @width - resulting_position.x) +
+      min(resulting_position.y, @height - resulting_position.y)
+  end
```

Comment:
You need to use `Array#min` here. (Thank you for this Ruby kit btw :-)

```rb
    return [resulting_position.x, @width - resulting_position.x].min +
      [resulting_position.y, @height - resulting_position.y].min
```

Reply:
Thank you! fixed now!

Reply:
Good catch. Just ran into this with the old version.